logaggregator/server.go

```diff
@@ -0,0 +1,349 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"strconv"
+	"sync"
+
+	"github.com/flynn/flynn/discoverd/client"
+	"github.com/flynn/flynn/logaggregator/snapshot"
+	"github.com/flynn/flynn/pkg/connutil"
+	"github.com/flynn/flynn/pkg/stream"
+	"github.com/flynn/flynn/pkg/syslog/rfc5424"
+	"github.com/flynn/flynn/pkg/syslog/rfc6587"
+
+	"github.com/flynn/flynn/Godeps/_workspace/src/gopkg.in/inconshreveable/log15.v2"
+)
+
+type Server struct {
+	*Aggregator
+	*Replicator
+
+	ll, rl, al net.Listener   // syslog, replication, and api listeners
+	lwg, rwg   sync.WaitGroup // syslog & replication wait groups
+
+	discd  *discoverd.Client
+	hb     discoverd.Heartbeater
+	srv    discoverd.Service
+	stream stream.Stream
+	eventc <-chan *discoverd.Event
+
+	api      http.Handler
+	shutdown chan struct{}
+}
+
+type ServerConfig struct {
+	syslogAddr, replicationAddr, apiAddr string
+
+	serviceName string
+	discoverd   *discoverd.Client
```

Comment:
Any fields that are expected to be set before calling `NewServer` should be capitalized/exported.